Practical 13, Exercise 3

(a) If the line print(print_max()) is included the result is given as:

    The value of the function print_max ()
    None

    Example of code:

    Enter a number: 5
    Enter a number: 8
    The largest number between 5.0 and 8.0 is 8.0
    None

    print_max function executes as expected.
    When print_max is finished executing its return statement has an absence of a value.
    In python an absence of a value is replaced with a None type.

    The print statement takes a None type as an argument and prints it to the console.
    as `None`

(b) There are two places where the () can be left out of the function print_max:
    
    The function definition:
    If this is left out of defining the function, a syntax error is given.
    () is needed after a function to execute the function and return the value

    The function invocation:
    If it is left out of the final print_max function there will not be a syntax error
    It won't return the value
    This is returning the method but not the value from the function
    It return RESTART

(c) If you included the line print(print_max):
    
    It returns: '<function print_max at 0x10d4016a8>'
    
    Instead of returning the function, the print is returning the functions address in memory
